distributed/utils.py

```diff
@@ -986,3 +986,14 @@ def format_time(n):
     if n >= 1e-3:
         return '%.2f ms' % (n * 1e3)
     return '%.2f us' % (n * 1e6)
+
+
+class DequeHandler(logging.Handler):
+    """ A logging.Handler that records records into a deque """
+    def __init__(self, *args, **kwargs):
+        n = kwargs.pop('n', 10000)
+        self.deque = deque(maxlen=n)
+        super(DequeHandler, self).__init__(*args, **kwargs)
+
+    def emit(self, record):
+        self.deque.append(record)
```

Comment:
@pitrou do you know of a cleaner or more standard way to do this?

Reply:
Perhaps you want to subclass [QueueHandler](https://docs.python.org/3/library/logging.handlers.html#logging.handlers.QueueHandler), or copy and adapt its implementation (it's quite simple). It allows to return structured yet picklable information.